lib/src/postgrest.dart

```diff
@@ -32,9 +32,11 @@ class PostgrestClient {
   }
 
   /// Perform a table operation.
-  PostgrestQueryBuilder from(String table) {
+  ///
+  /// [T] can only be [PostgrestList], [PostgrestMap], [PostgrestMap?], [PostgrestListResponse] or [PostgrestMapResponse]
+  PostgrestQueryBuilder<T> from<T>(String table) {
```

Comment:
We can probably add [this](https://github.com/supabase-community/supabase-flutter/issues/252#issuecomment-1288382397) suggestion here!

```suggestion
  PostgrestQueryBuilder<T> from<T extends Object?>(String table) {
```

Reply:
Also, how about adding assertion here with a helpful error message here so that users can only pass the 5 allowed types here.

```dart
assert(T == PostgrestList || T == PostgrestMap || T == Nullable<PostgrestMap> || T == PostgrestListResponse || T == PostgrestMapResponse);
```

I think we can make `_Nullable` public and hide it from the export.

Reply:
Adding an assertion is a good idea!